### PR TITLE
[sc-12991] Upgrade to openRTB 2.6: Fork new repo

### PR DESCRIPTION
**Pull request by Araknyfe**
Changes are mentionned in different releases. 
Mostly these are changes that we made on our old fork and that were not in the source repo.

### CODE REVIEW

**Review by GodefroiRoussel — COMMENTED** [review]
I think we have to change the Value in the [native1/response/data.go](https://github.com/onfocusio/openrtb-adagio/blob/main/native1/response/data.go#L57), previously we had 
`Value interface{} `json:"value"``

I'm wondering but I haven't verify yet, do we need to change [openrtb3/no_bid_reason.go](https://github.com/onfocusio/openrtb-adagio/blob/main/openrtb3/no_bid_reason.go#L4) to be an int as it was in our previous version ?

File: openrtb2/native.go [hunk 1]
```diff
@@ -22,6 +23,23 @@ import (
...
 type Native struct {
 
+	// Attribute:
+	//   request_native
+	// Type:
+	//   object; required
+	// Description:
+	// 		Native Struct for BidSwitch
+	//		https://protocol.bidswitch.com/ssp-protocol/ssp-native-request.html
+	RequestNative *native1.Request `json:"request_native"`
```
Comment (GodefroiRoussel) [inline openrtb2/native.go]: Add the omitempty

Comment (GodefroiRoussel) [reply openrtb2/native.go]: And also Line 60: 
`Request string 'json:"request"'` we need to add the omitempty

File: openrtb2/creative_attribute.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package openrtb2
```
Comment (GodefroiRoussel) [inline openrtb2/creative_attribute.go]: I have the impression that we are not using this file and that it is a duplicate of [adcom1/creative_attribute.go](https://github.com/onfocusio/openrtb-adagio/blob/main/adcom1/creative_attribute.go), if not we should delete it

File: openrtb2/audio.go [hunk 3]
```diff
@@ -66,6 +66,9 @@ type Audio struct {
 	//   OpenRTB <=2.5 defined only protocols 1..10.
 	Protocols []adcom1.MediaCreativeSubtype `json:"protocols,omitempty"`
 
+	//+       //   Array of supported audio protocols. Refer to List 5.8.
+	//+       Protocols []Protocol `json:"protocols,omitempty"`  @todo(athenais): check how to impl that
```
Comment (GodefroiRoussel) [inline openrtb2/audio.go]: You've got a todo here

**Comment by Araknyfe** [comment]
> I think we have to change the Value in the [native1/response/data.go](https://github.com/onfocusio/openrtb-adagio/blob/main/native1/response/data.go#L57), previously we had `Value interface{} `json:"value"``
> 
> I'm wondering but I haven't verify yet, do we need to change [openrtb3/no_bid_reason.go](https://github.com/onfocusio/openrtb-adagio/blob/main/openrtb3/no_bid_reason.go#L4) to be an int as it was in our previous version ?

Yes for Value good catch, as for NoBidReason I don't think so as it used once and cast as an int32 but it's worth checking again :)

File: openrtb2/audio.go (hunk 3 above)
Comment (Araknyfe) [reply openrtb2/audio.go]: Oops yes I forgot to remove when completed thanks

**Review by GodefroiRoussel — APPROVED** [review]

**Comment by Abyfall** [comment]
> I think we have to change the Value in the [native1/response/data.go](https://github.com/onfocusio/openrtb-adagio/blob/main/native1/response/data.go#L57), previously we had `Value interface{} `json:"value"``
> 
> I'm wondering but I haven't verify yet, do we need to change [openrtb3/no_bid_reason.go](https://github.com/onfocusio/openrtb-adagio/blob/main/openrtb3/no_bid_reason.go#L4) to be an int as it was in our previous version ?

Do you know why we chose to go from string to interface here ?

**Review by Abyfall — APPROVED** [review]

**Comment by GodefroiRoussel** [comment]
> > I think we have to change the Value in the [native1/response/data.go](https://github.com/onfocusio/openrtb-adagio/blob/main/native1/response/data.go#L57), previously we had `Value interface{} `json:"value"``
> > I'm wondering but I haven't verify yet, do we need to change [openrtb3/no_bid_reason.go](https://github.com/onfocusio/openrtb-adagio/blob/main/openrtb3/no_bid_reason.go#L4) to be an int as it was in our previous version ?
> 
> Do you know why we chose to go from string to interface here ?

When looking at the commit doing this change it is for : 

> Use interface for retro compatibility with Native 1.0

**Pull request merged**